modules/products/nodejs/src/main/kotlin/com/github/l34130/mise/runconfig/NodeRunConfigurationExtension.kt

```diff
@@ -1,46 +1,73 @@
 import com.intellij.javascript.nodejs.execution.runConfiguration.AbstractNodeRunConfigurationExtension
 import com.intellij.javascript.nodejs.execution.runConfiguration.NodeRunConfigurationLaunchSession
+import com.intellij.notification.NotificationType
+import com.intellij.openapi.diagnostic.Logger
 import com.intellij.openapi.options.SettingsEditor
 import com.jetbrains.nodejs.run.NodeJsRunConfiguration
 import org.jdom.Element
 
 class NodeRunConfigurationExtension : AbstractNodeRunConfigurationExtension() {
-    override fun getEditorTitle(): String = RunConfigurationSettingsEditor.EDITOR_TITLE
+    companion object {
+        private val LOG = Logger.getInstance(NodeRunConfigurationExtension::class.java)
+    }
+
+    override fun getEditorTitle(): String = MiseConfigurationPanelEditor.EDITOR_TITLE
 
     override fun <P : AbstractNodeTargetRunProfile> createEditor(configuration: P): SettingsEditor<P> =
-        RunConfigurationSettingsEditor(configuration)
+        MiseConfigurationPanelEditor(configuration)
 
-    override fun getSerializationId(): String = RunConfigurationSettingsEditor.SERIALIZATION_ID
+    override fun getSerializationId(): String = MiseConfigurationPanelEditor.SERIALIZATION_ID
 
     override fun readExternal(
         runConfiguration: AbstractNodeTargetRunProfile,
         element: Element,
     ) {
-        RunConfigurationSettingsEditor.readExternal(runConfiguration, element)
+        MiseConfigurationPanelEditor.readExternal(runConfiguration, element)
     }
 
     override fun writeExternal(
         runConfiguration: AbstractNodeTargetRunProfile,
         element: Element,
     ) {
-        RunConfigurationSettingsEditor.writeExternal(runConfiguration, element)
+        MiseConfigurationPanelEditor.writeExternal(runConfiguration, element)
     }
 
     override fun createLaunchSession(
         configuration: AbstractNodeTargetRunProfile,
         environment: ExecutionEnvironment,
     ): NodeRunConfigurationLaunchSession? {
-        val config = configuration as NodeJsRunConfiguration
+        if (!MiseConfigurationPanelEditor.isMiseEnabled(configuration)) {
+            return null
+        }
+
+        try {
+            when (configuration) {
+                is NodeJsRunConfiguration -> {
+                    val envs = configuration.envs.toMutableMap()
+                    envs.putAll(
+                        MiseCmd.loadEnv(
+                            workDir = configuration.workingDirectory,
+                            miseProfile = MiseConfigurationPanelEditor.getMiseProfile(configuration),
+                        )
+                    )
+                    configuration.envs = envs
+                }
 
-        val envs = config.envs.toMutableMap()
-        envs.putAll(MiseCmd.loadEnv(config.workingDirectory))
-        config.envs = envs
+                else -> {
+                    LOG.warn("Unsupported configuration type: ${configuration.javaClass}")
+                }
```

Comment:
👍 I've found some issues that Linux-based IDEA's RunConfiguration implemented class is NpmRunConfiguration.

Reply:
Same, for the npm one, I am not sure how it should work. I want to check how other env file plugins do it. Maybe it's possible to implement this generically for all languages